### ISSUE TITLE
consistency check for 15.7 and 12

### ISSUE THREAD

**Issue by karolszk** [phoughton/grim_test#2]
Hi! 

for mean 15.7 and n=12 consistency check say that is all consistent. Please take a look on the following output:

In [22]: mean_tester.summary_consistency_check('15.7','12')
Out[22]: 
{'ROUND_CEILING': True,
 'ROUND_DOWN': True,
 'ROUND_FLOOR': True,
 'ROUND_HALF_DOWN': True,
 'ROUND_HALF_EVEN': True,
 'ROUND_HALF_UP': True,
 'ROUND_UP': True,
 'ROUND_05UP': True}

but according to article:

https://www.sciencemag.org/news/2018/02/meet-data-thugs-out-expose-shoddy-and-questionable-research

at paragraph:

"For example, it might seem plausible if a researcher running a study involving 12 children aged 11 to 17 reports a mean age of 15.7. (After all, it’s less than 17 and greater than 11.) But the GRIM test reveals that value is mathematically impossible, because 15.7 is not a number that can be produced by dividing the sum of the ages by 12."

not.

**Comment by phoughton** [comment]
Thanks @karolszk I'll take a look into this.

**Comment by phoughton** [comment]
Hello @karolszk,

I had a look into your example, If we take the mean of these twelve numbers:
11,14,15,16,16,16,16,16,17,17,17,17
That gives us 15.666666....

This can be rounded to 15.7 (by ROUND_HALF_UP for example)

I suspect the author means (_but does not write_) 15.7**0**, e.g.:
```
mean_tester.summary_consistency_check('15.70','12')
```
Note the zero at the end, Which returns:
```
{'ROUND_05UP': False,
 'ROUND_CEILING': False,
 'ROUND_DOWN': False,
 'ROUND_FLOOR': False,
 'ROUND_HALF_DOWN': False,
 'ROUND_HALF_EVEN': False,
 'ROUND_HALF_UP': False,
 'ROUND_UP': False}
```

And this is inline with the [PrePubMed GRIM tool](http://www.prepubmed.org/grim_test/?direction=Up&mean=15.70&size=12) (That the author may have used) The PrePubMed tool is designed for 2 decimal places.

So, in summary, the GRIM library appears to be correct here, and the author probably means 15.70 and not 15.7.

Thanks

**Comment by karolszk** [comment]
Thank you @phoughton for fast check and answer. Now is clear for me. Indeed intention for 2 decimal points is explained in genuine GRIM-test paper "The GRIM test: A simple technique detects numerous anomalies in the reporting of results in psychology":

_"As psychological journals typically require 186 the reporting of means to 2 decimal places, the sample size corresponding to each mean typically needs to be less than 100 in order for its consistency to be checked."_

https://peerj.com/preprints/2064v1/
https://doi.org/10.7287/peerj.preprints.2064v1
